### ISSUE TITLE
defer registerRoot call

### ISSUE THREAD

**Issue by wcandillon** [remotion-dev/remotion#1050]
## Use Case

We are using remotion and webassembly together. Ideally we would like to wait for the web assembly to be loaded because even invoking. It would look like this:

```ts
LoadSkia().then(async () => {
  const RemotionVideo = (await import("./src/RemotionVideo")).default;
  registerRoot(RemotionVideo);
});
```

but it looks like the `registerRoot` call cannot be deferred. We are curious to find the best practice there.

**Comment by JonnyBurger** [comment]
React only supports default export for React.lazy: 

> React.lazy takes a function that must call a dynamic import(). This must return a Promise which resolves to a module with a default export containing a React component.

(https://reactjs.org/docs/code-splitting.html)

I'll look into whether it's possible to defer registerRoot()!

**Comment by wcandillon** [comment]
this recipe is awesome ❤️ Thank you for that, this is actually pretty useful.
fixes #1051

**Comment by Iamshankhadeep** [comment]
fixed by #1051

**Comment by wcandillon** [comment]
Just tried it, and I love it, really cool, thank you
